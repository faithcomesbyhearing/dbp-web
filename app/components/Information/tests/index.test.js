import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Information } from '..';
import {
  copyrights,
  invalidCopyrights,
  invalidCopyrights2,
} from '../../../utils/testUtils/copyrightData';

jest.mock('react-intl', () => ({
  	FormattedMessage: ({ defaultMessage }) => <span>{defaultMessage}</span>, // eslint-disable-line react/prop-types
  	defineMessages: (messages) => messages,
}));

describe('<Information /> component tests', () => {
	it('should match default snapshot', () => {
		const { container } = render(<Information copyrights={copyrights} />);
		expect(container).toMatchSnapshot();
	});

	it('should match snapshot with missing information', () => {
		const { container } = render(<Information copyrights={invalidCopyrights} />);
		expect(container).toMatchSnapshot();
	});

	it('should match snapshot with missing organizations', () => {
		const { container } = render(<Information copyrights={invalidCopyrights2} />);
		expect(container).toMatchSnapshot();
	});

	test('should open and close the information section when the button is clicked', () => {
		// Render the Information component
		const { container } = render(<Information copyrights={copyrights} />);

		// Query the toggle button by its text (provided by FormattedMessage)
		const toggleButton = screen.getByRole('button', { name: /learn more/i });

		const contentSection = container.querySelector('.copyrights-section');
		expect(contentSection).toHaveStyle('max-height: 0');

		// Click the button to open the section
		fireEvent.click(toggleButton);

		// Assert that the content section is now visible with the appropriate height
		expect(contentSection).toHaveStyle('max-height: 515px');

		// Assert that the arrow icon rotates
		const icon = container.querySelector('svg');
		expect(icon).toBeInTheDocument();

		expect(icon).toHaveClass('rotate');

		// Click the button again to close the section
		fireEvent.click(toggleButton);

		// Assert that the content section is closed again
		expect(contentSection).toHaveStyle('max-height: 0');

		// Ensure the icon no longer has the 'rotate' class
		expect(icon).not.toHaveClass('rotate');
	});
});
