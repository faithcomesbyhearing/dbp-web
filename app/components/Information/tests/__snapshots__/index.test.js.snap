// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Information /> component tests should match default snapshot 1`] = `
<div>
  <section
    class="information"
  >
    <button
      class="information-toggle"
      type="button"
    >
      <span>
        Copyrighted Material
      </span>
       | 
      <span
        class="learn-more"
      >
        <span>
          Learn More
        </span>
      </span>
      <svg
        class="icon"
      >
        <use
          xlink:href="/svglist.svg#arrow_down"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </button>
    <div
      aria-hidden="true"
      class="rah-static rah-static--height-zero "
      style="height: 0px; overflow: hidden;"
    >
      <div
        style="display: none;"
      >
        <div
          class="copyrights-section"
          style="max-height: 0;"
        >
          <div
            class="ot-copyright"
          >
            <div
              class="cp-section"
            >
              <h3>
                <span>
                  This Bible audio is brought to you by:
                </span>
              </h3>
              <a
                href="http://www.fcbh.org"
                rel="noopener"
                target="_blank"
              >
                <img
                  alt="Copyright owners logo: Faith Comes By Hearing"
                  class="image-landscape fcbh-copyright-logo"
                  src="https://bible.cloud/images/logo/FCBH_Color.svg"
                />
              </a>
              <p>
                <span>
                  Text: The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
                </span>
                <br />
                <span>
                  English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
                </span>
                <br />
                <span>
                  Audio: ℗ 2009 Hosanna
                </span>
                <br />
              </p>
            </div>
            <div
              class="cp-section"
            >
              <h3>
                <span>
                  This Bible text is brought to you by:
                </span>
              </h3>
              <a
                href="https://www.crossway.org"
                rel="noopener"
                target="_blank"
              >
                <img
                  alt="Copyright owners logo: Crossway"
                  class="image-landscape"
                  src="https://images.bible.cloud/logo/crossway.svg"
                />
              </a>
              <h3>
                <span>
                  This Bible text is brought to you by:
                </span>
              </h3>
              <a
                class="org-name"
                rel="noopener"
                target="_blank"
              >
                Good News Publishers
              </a>
              <p>
                <span>
                  The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
                </span>
                <br />
                <span>
                  English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
                </span>
                <br />
              </p>
            </div>
          </div>
          <div
            class="nt-copyright"
          >
            <div
              class="cp-section"
            >
              <h3>
                <span>
                  This Bible audio is brought to you by:
                </span>
              </h3>
              <a
                href="http://www.fcbh.org"
                rel="noopener"
                target="_blank"
              >
                <img
                  alt="Copyright owners logo: Faith Comes By Hearing"
                  class="image-landscape fcbh-copyright-logo"
                  src="https://bible.cloud/images/logo/FCBH_Color.svg"
                />
              </a>
              <p>
                <span>
                  Text: The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
                </span>
                <br />
                <span>
                  English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
                </span>
                <br />
                <span>
                  Audio: ℗ 2009 Hosanna
                </span>
                <br />
              </p>
            </div>
            <div
              class="cp-section"
            >
              <h3>
                <span>
                  This Bible text is brought to you by:
                </span>
              </h3>
              <a
                href="https://www.crossway.org"
                rel="noopener"
                target="_blank"
              >
                <img
                  alt="Copyright owners logo: Crossway"
                  class="image-landscape"
                  src="https://images.bible.cloud/logo/crossway.svg"
                />
              </a>
              <h3>
                <span>
                  This Bible text is brought to you by:
                </span>
              </h3>
              <a
                class="org-name"
                rel="noopener"
                target="_blank"
              >
                Good News Publishers
              </a>
              <p>
                <span>
                  The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
                </span>
                <br />
                <span>
                  English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
                </span>
                <br />
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ministry-statement"
    >
      <div
        class="ministry-tagline"
      >
        <span>
          ©
        </span>
         
        <a
          href="http://www.bible.is"
          rel="noopener"
          target="_blank"
        >
          <span>
            Bible.is
          </span>
        </a>
        <span>
          , a ministry of
        </span>
         
        <a
          href="https://www.faithcomesbyhearing.com"
          rel="noopener"
          target="_blank"
        >
          <span>
            Faith Comes By Hearing
          </span>
        </a>
         
        <span>
          ®.
        </span>
         
      </div>
      <div
        class="ministry-terms-support"
      >
        <a
          href="/terms"
          id="terms-of-service"
        >
          <span>
            Terms and Conditions
          </span>
        </a>
          |  
        <a
          href="https://support.bible.is/contact"
          id="help-support"
          rel="noopener"
          target="_blank"
        >
          <span>
            Help & Support
          </span>
        </a>
      </div>
    </div>
  </section>
</div>
`;

exports[`<Information /> component tests should match snapshot with missing information 1`] = `
<div>
  <section
    class="information"
  >
    <button
      class="information-toggle"
      type="button"
    >
      <span>
        Copyrighted Material
      </span>
       | 
      <span
        class="learn-more"
      >
        <span>
          Learn More
        </span>
      </span>
      <svg
        class="icon"
      >
        <use
          xlink:href="/svglist.svg#arrow_down"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </button>
    <div
      aria-hidden="true"
      class="rah-static rah-static--height-zero "
      style="height: 0px; overflow: hidden;"
    >
      <div
        style="display: none;"
      >
        <div
          class="copyrights-section"
          style="max-height: 0;"
        >
          <div
            class="ot-copyright"
          >
            <div
              class="cp-section"
            >
              <p>
                <span>
                  Text: The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
                </span>
                <br />
                <span>
                  English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
                </span>
                <br />
                <span>
                  Audio: ℗ 2009 Hosanna
                </span>
                <br />
              </p>
            </div>
            <div
              class="cp-section"
            >
              <p>
                <span>
                  The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
                </span>
                <br />
                <span>
                  English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
                </span>
                <br />
              </p>
            </div>
          </div>
          <div
            class="nt-copyright"
          >
            <div
              class="cp-section"
            >
              <p>
                <span>
                  Text: The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
                </span>
                <br />
                <span>
                  English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
                </span>
                <br />
                <span>
                  Audio: ℗ 2009 Hosanna
                </span>
                <br />
              </p>
            </div>
            <div
              class="cp-section"
            >
              <p>
                <span>
                  The ESV Bible® (The Holy Bible, English Standard Version®) Copyright © 2001 by Crossway, a publishing ministry of Good News Publishers. ESV® Text Edition: 2007. All rights reserved.
                </span>
                <br />
                <span>
                  English Standard Version, ESV, and the ESV logo are registered trademarks of Good News Publishers.  Used by permission.
                </span>
                <br />
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ministry-statement"
    >
      <div
        class="ministry-tagline"
      >
        <span>
          ©
        </span>
         
        <a
          href="http://www.bible.is"
          rel="noopener"
          target="_blank"
        >
          <span>
            Bible.is
          </span>
        </a>
        <span>
          , a ministry of
        </span>
         
        <a
          href="https://www.faithcomesbyhearing.com"
          rel="noopener"
          target="_blank"
        >
          <span>
            Faith Comes By Hearing
          </span>
        </a>
         
        <span>
          ®.
        </span>
         
      </div>
      <div
        class="ministry-terms-support"
      >
        <a
          href="/terms"
          id="terms-of-service"
        >
          <span>
            Terms and Conditions
          </span>
        </a>
          |  
        <a
          href="https://support.bible.is/contact"
          id="help-support"
          rel="noopener"
          target="_blank"
        >
          <span>
            Help & Support
          </span>
        </a>
      </div>
    </div>
  </section>
</div>
`;

exports[`<Information /> component tests should match snapshot with missing organizations 1`] = `
<div>
  <section
    class="information"
  >
    <button
      class="information-toggle"
      type="button"
    >
      <span>
        Copyrighted Material
      </span>
       | 
      <span
        class="learn-more"
      >
        <span>
          Learn More
        </span>
      </span>
      <svg
        class="icon"
      >
        <use
          xlink:href="/svglist.svg#arrow_down"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        />
      </svg>
    </button>
    <div
      aria-hidden="true"
      class="rah-static rah-static--height-zero "
      style="height: 0px; overflow: hidden;"
    >
      <div
        style="display: none;"
      >
        <div
          class="copyrights-section"
          style="max-height: 0;"
        >
          <div
            class="ot-copyright"
          >
            <div
              class="cp-section"
            >
              <h3>
                <span>
                  This Bible audio is brought to you by:
                </span>
              </h3>
              <a
                href="https://www.crossway.org"
                rel="noopener"
                target="_blank"
              >
                <img
                  alt="Copyright owners logo: Crossway"
                  class="image-landscape fcbh-copyright-logo"
                  src="https://images.bible.cloud/logo/crossway.svg"
                />
              </a>
              <h3>
                <span>
                  This Bible audio is brought to you by:
                </span>
              </h3>
              <a
                class="org-name"
                rel="noopener"
                target="_blank"
              >
                Good News Publishers
              </a>
            </div>
            <div
              class="cp-section"
            >
              <h3>
                <span>
                  This Bible text is brought to you by:
                </span>
              </h3>
              <a
                href="https://www.crossway.org"
                rel="noopener"
                target="_blank"
              >
                <img
                  alt="Copyright owners logo: Crossway"
                  class="image-landscape"
                  src="https://images.bible.cloud/logo/crossway.svg"
                />
              </a>
              <h3>
                <span>
                  This Bible text is brought to you by:
                </span>
              </h3>
              <a
                class="org-name"
                rel="noopener"
                target="_blank"
              >
                Good News Publishers
              </a>
            </div>
          </div>
          <div
            class="nt-copyright"
          >
            <div
              class="cp-section"
            >
              <h3>
                <span>
                  This Bible audio is brought to you by:
                </span>
              </h3>
              <a
                href="https://www.crossway.org"
                rel="noopener"
                target="_blank"
              >
                <img
                  alt="Copyright owners logo: Crossway"
                  class="image-landscape fcbh-copyright-logo"
                  src="https://images.bible.cloud/logo/crossway.svg"
                />
              </a>
              <h3>
                <span>
                  This Bible audio is brought to you by:
                </span>
              </h3>
              <a
                class="org-name"
                rel="noopener"
                target="_blank"
              >
                Good News Publishers
              </a>
            </div>
            <div
              class="cp-section"
            >
              <h3>
                <span>
                  This Bible text is brought to you by:
                </span>
              </h3>
              <a
                href="https://www.crossway.org"
                rel="noopener"
                target="_blank"
              >
                <img
                  alt="Copyright owners logo: Crossway"
                  class="image-landscape"
                  src="https://images.bible.cloud/logo/crossway.svg"
                />
              </a>
              <h3>
                <span>
                  This Bible text is brought to you by:
                </span>
              </h3>
              <a
                class="org-name"
                rel="noopener"
                target="_blank"
              >
                Good News Publishers
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ministry-statement"
    >
      <div
        class="ministry-tagline"
      >
        <span>
          ©
        </span>
         
        <a
          href="http://www.bible.is"
          rel="noopener"
          target="_blank"
        >
          <span>
            Bible.is
          </span>
        </a>
        <span>
          , a ministry of
        </span>
         
        <a
          href="https://www.faithcomesbyhearing.com"
          rel="noopener"
          target="_blank"
        >
          <span>
            Faith Comes By Hearing
          </span>
        </a>
         
        <span>
          ®.
        </span>
         
      </div>
      <div
        class="ministry-terms-support"
      >
        <a
          href="/terms"
          id="terms-of-service"
        >
          <span>
            Terms and Conditions
          </span>
        </a>
          |  
        <a
          href="https://support.bible.is/contact"
          id="help-support"
          rel="noopener"
          target="_blank"
        >
          <span>
            Help & Support
          </span>
        </a>
      </div>
    </div>
  </section>
</div>
`;
